MAUI/DataGrid/Editing.md

```diff
@@ -0,0 +1,376 @@
+            this.storedValues = null;
+        }
+        Debug.WriteLine("End Edit Called");
+    }
+
+    protected Dictionary<string, object> BackUp()
+    {
+        var dictionary = new Dictionary<string, object>();
+        var itemProperties = this.GetType().GetTypeInfo().DeclaredProperties;
+        foreach (var pDescriptor in itemProperties)
+        {
+            if (pDescriptor.CanWrite)
+                dictionary.Add(pDescriptor.Name, pDescriptor.GetValue(this));
+        }
+        return dictionary;
+    }
+
+    #endregion
+}
+{% endhighlight %}
+
+
+## Editing events
+
+The SfDataGrid triggers the following events while editing:
+
+### CurrentCellBeginEdit
+
+The [SfDataGrid.CurrentCellBeginEdit]() event occurs when the CurrentCell enters into edit mode. The [DataGridCurrentCellBeginEditEventArgs]() has the following members which provides information for `SfDataGrid.CurrentCellBeginEdit` event:
+
+ * [Cancel](): When this member set to ‘true’, the event is canceled and the CurrentCell does not enter into the edit mode.
+ * [RowColumnIndex](): Gets the current row and column index of the DataGrid.
+ * [Column](): Gets the Grid Column of the SfDataGrid.
+
+To hook the `SfDataGrid.CurrentCellBeginEdit` event, follow the code example:
+
+{% highlight c# %}
+this.dataGrid.CurrentCellBeginEdit += DataGrid_CurrentCellBeginEdit;
+
+private void DataGrid_CurrentCellBeginEdit(object sender, DataGridCurrentCellBeginEditEventArgs e)
+{
+    // Editing prevented for the cell at RowColumnIndex(2,2).
+    if (e.RowColumnIndex == new Syncfusion.Maui.GridCommon.ScrollAxis.RowColumnIndex(2, 2)) 
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+### CurrentCellEndEdit
+
+The [CurrentCellEndEdit]() event occurs when the CurrentCell exits the edit mode. The [DataGridCurrentCellEndEditEventArgs]() has following members which provides information for `SfDataGrid.CurrentCellEndEdit` event:
+
+ * [RowColumnIndex](): Gets the current row and column index of the DataGrid.
+ * [Cancel](): When this member set to ‘true’, the event is canceled and the edited value is not committed in the underlying collection.
+
+To hook the `SfDataGrid.CurrentCellEndEdit` event, follow the code example:
+
+{% highlight c# %}
+this.dataGrid.CurrentCellEndEdit += DataGrid_CurrentCellEndEdit;
+
+private void DataGrid_CurrentCellEndEdit(object sender, DataGridCurrentCellEndEditEventArgs e)
+{
+    // Editing prevented for the cell at RowColumnIndex(1,3).
+    if (e.RowColumnIndex == new Syncfusion.Maui.GridCommon.ScrollAxis.RowColumnIndex(1, 3))
+        e.Cancel = true;
+}
+
+{% endhighlight %}
+
+## Programmatically edit a cell
+
+### Begin editing
+
+The SfDataGrid allows you to programmatically edit a cell by calling the [SfDataGrid.BeginEdit]() method. By calling this method, the particular cell enters into edit mode, allowing you to edit the data manually or programmatically. To programmatically edit a cell, refer to the following code example:
```

Comment:
The SfDataGrid allows you to programmatically edit a cell by calling the SfDataGrid.BeginEdit() method. This method enters the particular cell into edit mode, enabling the editing of data programmatically. To programmatically edit a cell, refer to the code example below:

Reply:
![image](https://github.com/syncfusion-content/maui-docs/assets/113881396/68622607-f559-4409-930a-4de93d6a4776)